engine/src/main/java/org/terasology/world/block/BlockRegionc.java

```diff
@@ -0,0 +1,799 @@
+     * @param dest destination; will hold the result
+     * @return {@code dest} (after modification)
+     */
+    BlockRegion translate(int dx, int dy, int dz, BlockRegion dest);
+
+    /**
+     * Translate this region by the given vector {@code vec}.
+     *
+     * @param vec the vector to translate by
+     * @param dest destination; will hold the result
+     * @return {@code dest} (after modification)
+     */
+    default BlockRegion translate(Vector3ic vec, BlockRegion dest) {
+        return this.translate(vec.x(), vec.y(), vec.z(), dest);
+    }
+
+    // -- expand -----------------------------------------------------------------------------------------------------//
+
+    /**
+     * Expand this region by adding the given {@code extents} for each face of the region.
+     *
+     * @param dx the amount of blocks to extend this region by along the x axis in both directions
+     * @param dy the amount of blocks to extend this region by along the y axis in both directions
+     * @param dz the amount of blocks to extend this region by along the z axis in both directions
+     * @param dest destination; will hold the result
+     * @return {@code dest} (after modification)
+     * @throws IllegalArgumentException if extending this region would result in any non-positive dimension
+     */
+    BlockRegion expand(int dx, int dy, int dz, BlockRegion dest);
+
+    /**
+     * Expand this region by adding the given {@code extents} for each face of a region.
+     *
+     * @param vec the amount of blocks to expand this region by
+     * @param dest destination; will hold the result
+     * @return {@code dest} (after modification)
+     * @throws IllegalArgumentException if extending this region would result in any non-positive dimension
+     */
+    default BlockRegion expand(Vector3ic vec, BlockRegion dest) {
+        return this.expand(vec.x(), vec.y(), vec.z(), dest);
+    }
+
+    /**
+     * Extend this region by adding the given {@code extents} for each face of a region.
+     * <p>
+     * The extents will be floored for each dimension.
+     *
+     * @param dx the amount of blocks to extend this region by along the x axis in both directions
+     * @param dy the amount of blocks to extend this region by along the y axis in both directions
+     * @param dz the amount of blocks to extend this region by along the z axis in both directions
+     * @param dest destination; will hold the result
+     * @return {@code dest} (after modification)
+     * @throws IllegalArgumentException if extending this region would result in any non-positive dimension
+     */
+    //TODO: why do we offer float variants here?
+    default BlockRegion expand(float dx, float dy, float dz, BlockRegion dest) {
+        return this.expand(
+                Math.roundUsing(dx, RoundingMode.FLOOR),
+                Math.roundUsing(dy, RoundingMode.FLOOR),
+                Math.roundUsing(dz, RoundingMode.FLOOR),
+                dest);
+    }
```

Comment:
I'm not sure this is correct :thinking: I think we are missing the `0.5f` margin to each side, e.g., 
```java
// bounding box: (.5, .5, .5)..(1.5, 1.5, 1.5)
new BlockRegion(1,1,1).expand(0.5f, 0.5f, 0.5f) == new BlockRegion(0,0,0, 2,2,2)
```

```suggestion
        return this.expand(
                Math.roundUsing(dx + 0.5f, RoundingMode.FLOOR),
                Math.roundUsing(dy + 0.5f, RoundingMode.FLOOR),
                Math.roundUsing(dz + 0.5f, RoundingMode.FLOOR),
                dest);
    }
```

Reply:
I don't understand why we need `expand` to accept float values in the first place.
Can a `BlockRegion` contain only parts of blocks? If not I don't see a use case in which we would need to support float input.